### PR TITLE
Sha256

### PR DESCRIPTION
**Pull request by shradhakhatiwada**
#6 Implemented Custom SHA-256 Hashing Algorithm for Secure File Storage System

This pull request adds the core implementation of the SHA-256 hashing algorithm from scratch without using built-in libraries. Key highlights include:

- **Message Pre-processing (Padding):** Implemented padding and chunking of the input message to align with the SHA-256 specification.
- **Message Schedule Array Expansion:** Extended the message schedule array using bitwise operations.
- **Compression Function:** Added the main loop with operations like bitwise right rotation, message expansion, and updating the eight working variables (`a` through `h`).
- **Final Hash Generation:** Produced the final 256-bit (64 hex characters) hash by combining the working variables after processing all message blocks.
- **Commented Code for Clarity:** Added detailed comments to explain each function and formula used in the implementation.

This implementation will be used for password hashing and digital signatures in the "Secure File Storage System" project. The algorithm adheres to the official SHA-256 specification and improves security by ensuring proper hashing of sensitive data.

### CODE REVIEW

File: backend/src/crypto/sha.js
Comment (aashishxetri5) [inline backend/src/crypto/sha.js:1]: Remove unnecessary files.

File: backend/src/crypto/sha256.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    return H.map(value => value.toString(16).padStart(8, '0')).join('');
+}
+
+// Example usage
+const message = "securefile";
+const hash = sha256(message);
+console.log("SHA-256 Hash: ", hash);
+
+
+console.log("SHA-256 Hash: ", sha256("securefiles"));
```
Comment (aashishxetri5) [inline backend/src/crypto/sha256.js:131]: remove these `console.logs()` and example usage.

File: backend/src/crypto/sha256.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    return ((value >>> shift) | (value << (32 - shift))) >>> 0;
+}
+
+/**
+ * The main SHA-256 hash function.
+ * This function takes a message and computes its SHA-256 hash based on the algorithm specification.
+ * @param {string} message - The input message to be hashed
+ * @returns {string} - The resulting 256-bit hash as a hex string
+ */
+function sha256(message) {
```
Comment (aashishxetri5) [inline backend/src/crypto/sha256.js:39]: add this function to export.

**Comment by aashishxetri5** [comment]
Also, make sure you've written unit test cases and the methods pass them.

File: backend/src/crypto/sha256.js (hunk 2 above)
Comment (shradhakhatiwada) [reply backend/src/crypto/sha256.js:42]: Added the sha256 function to exports.

**Review by aashishxetri5 — APPROVED** [review]

**Pull request merged**